assets/app/scripts/directives/podStatusChart.js

```diff
@@ -134,41 +145,47 @@ angular.module('openshiftConsole')
           $scope.podStatusData = data.columns;
         }
 
-        function countPodPhases() {
-          var countByPhase = {};
-          var incrementCount = function(phase) {
-            countByPhase[phase] = (countByPhase[phase] || 0) + 1;
-          };
+        function isReady(pod) {
+          var numReady = numContainersReadyFilter(pod);
+          var total = pod.spec.containers.length;
 
-          var isReady = function(pod) {
-            var numReady = numContainersReadyFilter(pod);
-            var total = pod.spec.containers.length;
+          return numReady === total;
+        }
 
-            return numReady === total;
-          };
+        function getPhase(pod) {
+          if (isTerminatingFilter(pod)) {
+            return 'Terminating';
+          }
 
-          angular.forEach($scope.pods, function(pod) {
-            // Count 'Warning' as its own phase, even if not strictly accurate,
-            // so it appears in the donut chart. Warnings are too important not
-            // to call out.
-            if (isTroubledPodFilter(pod)) {
-              incrementCount('Warning');
-              return;
-            }
+          if (isTroubledPodFilter(pod)) {
+            return 'Warning';
+          }
 
-            // Also count running, but not ready, as its own phase.
-            if (pod.status.phase === 'Running' && !isReady(pod)) {
-              incrementCount('Not Ready');
-              return;
-            }
+          if (isPullingImageFilter(pod)) {
+            return 'Pulling';
+          }
+
+          // Also count running, but not ready, as its own phase.
+          if (pod.status.phase === 'Running' && !isReady(pod)) {
+            return 'Not Ready';
+          }
+
+          return _.get(pod, 'status.phase', 'Unknown');
+        }
 
-            incrementCount(pod.status.phase);
+        function countPodPhases() {
+          var countByPhase = {};
+
+          angular.forEach($scope.pods, function(pod) {
+            var phase = getPhase(pod);
+            countByPhase[phase] = (countByPhase[phase] || 0) + 1;
           });
 
           return countByPhase;
         }
 
-        $scope.$watch(countPodPhases, updateChart, true);
+        var debounceUpdate = _.debounce(updateChart, 350, { maxWait: 500 });
```

Comment:
Approved.

On Sat, Apr 2, 2016 at 7:11 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/directives/podStatusChart.js
> <https://github.com/openshift/origin/pull/8335#discussion_r58301593>:
>
> >            });
> >
> >            return countByPhase;
> >          }
> >
> > -        $scope.$watch(countPodPhases, updateChart, true);
> > +        var debounceUpdate = _.debounce(updateChart, 350, { maxWait: 500 });
>
> Yeah, felt better to me without it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8335/files/6874ebfabfb2b02ac00416631a5ec79a89c8c40d#r58301593>
>